Signal/src/call/CallService.swift

```diff
@@ -388,11 +389,17 @@ protocol CallServiceObserver: class {
     public func handleMissedCall(_ call: SignalCall, thread: TSContactThread) {
         AssertIsOnMainThread()
         // Insert missed call record
-        let callRecord = TSCall(timestamp: NSDate.ows_millisecondTimeStamp(),
-                                withCallNumber: thread.contactIdentifier(),
-                                callType: RPRecentCallTypeMissed,
-                                in: thread)
-        callRecord.save()
+        if call.callRecord == nil {
```

Comment:
nit: Testing if a thing is nil, and then doing something with that thing is often a good spot for an `if let`. 

So, specifically, if you flip the condition you could do this:
```
if let callRecord = call.callRecord {
    callRecord.updateCallType(RPRecentCallTypeMissed)
} else {
    callRecord = TSCall(timestamp: NSDate.ows_millisecondTimeStamp(),
                                      withCallNumber: thread.contactIdentifier(),
                                      callType: RPRecentCallTypeMissed,
                                      in: thread)
}
```

The motivation being that code with less `!`'s is a little more resilient to future refactors, because the compiler will remind you if that thing is possibly nil rather than crashing at runtime if it is.